backend/site-api/src/db/site_repo.rs

```diff
@@ -219,19 +237,34 @@ impl SiteRepoTrait for SiteRepo {
         // Connect to the database
         let pool = SqlitePoolOptions::new().connect(&site_db_url).await?;
 
-        // Migrate the database
-        sqlx::migrate!("db/sites/migrations")
-            .run(&pool)
-            .await
-            .map_err(|e| DbError::Migrate(e.to_string()))?;
-
         // Insert the connection pool into the map.
         let mut db_pools = self.site_db_pools.write().await;
         db_pools.insert(id.to_string(), pool.clone());
 
+        self.migrate_db(&pool).await?;
+
         Ok(pool)
     }
 
+    async fn migrate_db(&self, pool: &SqlitePool) -> Result<(), DbError> {
+        // Migrate the database
+        sqlx::migrate!("db/sites/migrations")
+            .run(pool)
+            .await
+            .map_err(|e| DbError::Migrate(e.to_string()))?;
+        Ok(())
+    }
+
+    async fn migrate_all(&self, sites: Vec<SiteMetadataEntity>) -> Result<(), DbError> {
```

Comment:
I realized we didn't have a way to migrate each site DB, so I added some code for it. Now the migrations will be run when the app starts, and when a new site is created.